finality-aleph/src/party.rs

```diff
@@ -305,7 +315,6 @@ where
                 .collect();
         // wait for the first task returning true
         tasks.next().await;
-        //tasks.filter(|b| std::future::ready(*b)).next().await;
```

Comment:
I think we actually want `tasks.filter(|b| std::future::ready(*b)).next()` instead of `tasks.next()`. If we do not participate in the consensus, the `session_task` will complete immediately returning false.